src/Select/Select.js

```diff
@@ -0,0 +1,539 @@
+ */
+export default class Select extends Component<Props, State> {
+  static defaultProps = {
+    placeholder: 'Select...',
+    placement: 'bottom-start',
+    size: 'large'
+  };
+
+  state: State = {
+    highlightedIndex: this.props.defaultHighlightedIndex,
+    isOpen: Boolean(this.props.defaultIsOpen),
+    selectedItem: this.props.defaultSelectedItem
+  };
+
+  id: string = this.props.id || `select-${generateId()}`;
+
+  selectTrigger: ?React$Component<*, *>;
+
+  itemMatcher: any;
+
+  render() {
+    const {
+      data,
+      disabled,
+      modifiers,
+      name,
+      placeholder,
+      readOnly,
+      size,
+      variant,
+      triggerRef,
+      ...restProps
+    } = this.props;
+    const isOpen = this.getControllableValue('isOpen');
+    const selectedItem = this.getControllableValue('selectedItem');
+
+    const rootProps = {
+      id: this.id,
+      ...restProps,
+      data,
+      disabled: disabled || readOnly,
+      highlightedIndex: this.getHighlightedOrSelectedIndex(),
+      getMenuProps: this.getMenuProps,
+      getItemProps: this.getItemProps,
+      getTriggerProps: this.getTriggerProps,
+      isOpen,
+      modifiers: {
+        contentWidth: contentWidthModifier,
+        ...modifiers
+      },
+      onClose: this.close,
+      onOpen: this.open
+    };
+
+    const selectTriggerProps = {
+      disabled,
+      isOpen,
+      name,
+      placeholder,
+      readOnly,
+      size,
+      triggerRef: (node: ?React$Component<*, *>) => {
+        this.selectTrigger = node;
+        triggerRef && triggerRef(node);
+      },
+      item: selectedItem,
+      variant
+    };
+
+    return (
+      <Root {...rootProps}>
+        <SelectTrigger {...selectTriggerProps} />
+      </Root>
+    );
+  }
+
+  getMenuItemId = (index: string) => {
+    return `${this.id}-item-${index}`;
+  };
+
+  getTriggerProps = (props: Object = {}) => {
+    const { disabled, invalid, readOnly, required } = this.props;
+
+    return {
+      // Props set by caller, e.g. Dropdown
+      ...props,
+
+      // Props set by this component
+      'aria-haspopup': 'listbox',
+      'aria-invalid': invalid,
+      'aria-readonly': readOnly,
+      'aria-required': required,
+      disabled,
+      onKeyDown: !readOnly ? this.onTriggerKeyDown : undefined,
+      tabIndex: !disabled ? 0 : undefined
+    };
+  };
+
+  getMenuProps = (props: Object = {}) => {
+    return {
+      // Props set by caller, e.g. Dropdown
+      ...props,
+
+      // Props set by this component
+      role: 'listbox'
+    };
+  };
+
+  getItemProps = (props: Object = {}, scope: Object) => {
+    const { item } = scope;
+    const selectedItem = this.getControllableValue('selectedItem');
+
+    return {
+      // Props set by caller, e.g. Dropdown
+      ...props,
+
+      // Props set by this component
+      'aria-selected': selectedItem ? selectedItem.value === item.value : false,
+      onClick: this.onSelect.bind(null, item),
```

Comment:
This clobbers any `onClick` that a user might put on their items. Is that intentional or should we compose them?

Reply:
Should we support user defined onItem click? It is easy enough to compose. The user has this capability via onChange and onSelect handlers.

Reply:
I composed the functions, though I think for the majority of cases, using `onSelect` and/or `onChange` would be the preferred approach.